src/engine/filters/enginefiltermoogladder4.h

```diff
@@ -169,12 +169,12 @@ class EngineFilterMoogLadderBase : public EngineObjectConstIn {
                 double new2 = processSample(pIn[i+1], &m_buf[1]);
 
                 if (i < iBufferSize / 2) {
-                    pOutput[i] = old1;
-                    pOutput[i + 1] = old2;
+                    pOutput[i] = static_cast<CSAMPLE>(old1);
+                    pOutput[i + 1] = static_cast<CSAMPLE>(old2);
                 } else {
-                    pOutput[i] = new1 * cross_mix + old1 * (1.0 - cross_mix);
-                    pOutput[i + 1] = new2 * cross_mix
-                            + old2 * (1.0 - cross_mix);
+                    pOutput[i] = static_cast<CSAMPLE>(new1 * cross_mix + old1 * (1.0 - cross_mix));
```

Comment:
this can also become a float calculation

Reply:
How? I thought you said the filter needs double precision to sound good?

Reply:
Here this is true: 
static_cast<float>(double * float) == float * float

so cross_mix can become a CSAMPLE_GAIN

We need double precision in our filters whenever a value is stored and used in the following calculation. Than an accumulated error can be significant.

Reply:
But `cross_mix` is incremented by `cross_inc` in each iteration?

Reply:
Also, `cross_mix` is multiplies with `new1` (double), so it's actually double * double ?

Reply:
Yes you are right. The current version is already good.